src/nvim/strings.c

```diff
@@ -394,6 +394,18 @@ void del_trailing_spaces(char_u *ptr)
   }
 }
 
+#if !defined(HAVE_STRNLEN)
+size_t vim_strnlen(const char *s, size_t n)
```

Comment:
if we add this, by our conventions it should be `xstrnlen`

Reply:
Renamed to xstrnlen